### PR TITLE
Addressing Arch Board Feedback

### PR DESCRIPTION
**Pull request by yogeshmo**
- Adding AAD token support
- Adding overloads for email recipient
- Creating yml file for the live test pipeline
- Make the "to" property in EmailRecipients optional

### CODE REVIEW

File: sdk/communication/azure-communication-email/README.md [hunk 1]
```diff
@@ -88,6 +88,20 @@ EmailClient emailClient = new EmailClientBuilder()
     .buildClient();
 ```
 
+### Azure Active Directory Token Authentication
+A `DefaultAzureCredential` object must be passed to the `EmailClientBuilder` via the credential() function. Endpoint must also be set via the endpoint() function.
```
Comment (vcolin7) [inline sdk/communication/azure-communication-email/README.md:92]:
```suggestion
A `DefaultAzureCredential` object must be passed to the `EmailClientBuilder` via the `credential()` method. An endpoint must also be set via the `endpoint()` method.
```

File: sdk/communication/azure-communication-email/README.md [hunk 2]
```diff
@@ -88,6 +88,20 @@ EmailClient emailClient = new EmailClientBuilder()
...
+
+`AZURE_CLIENT_SECRET`, `AZURE_CLIENT_ID` and `AZURE_TENANT_ID` environment variables are needed to create a DefaultAzureCredential object.
```
Comment (vcolin7) [inline sdk/communication/azure-communication-email/README.md:94]:
```suggestion
The `AZURE_CLIENT_SECRET`, `AZURE_CLIENT_ID`, and `AZURE_TENANT_ID` environment variables are needed to create a `DefaultAzureCredential` object.
```

File: sdk/communication/azure-communication-email/src/main/java/com/azure/communication/email/EmailClientBuilder.java [hunk 3]
```diff
@@ -240,6 +263,15 @@ private AzureCommunicationServicesClientImpl buildInnerClient() {
         return client;
     }
 
+    private HttpPipelinePolicy createHttpPipelineAuthPolicy() {
+        if (this.tokenCredential != null) {
+            return new BearerTokenAuthenticationPolicy(
+                this.tokenCredential, "https://communication.azure.com//.default");
+        } else {
+            return new HmacAuthenticationPolicy(this.azureKeyCredential);
```
Comment (vcolin7) [inline sdk/communication/azure-communication-email/src/main/java/com/azure/communication/email/EmailClientBuilder.java:271]: What if this credential is also `null`? Do we check for that before calling this method?

**Review by vcolin7 — COMMENTED** [review]
Added some comments but in general the changes look good to me :)

File: sdk/communication/azure-communication-email/src/main/java/com/azure/communication/email/EmailClientBuilder.java (hunk 3 above)
Comment (yogeshmo) [reply sdk/communication/azure-communication-email/src/main/java/com/azure/communication/email/EmailClientBuilder.java:274]: Good point, I'll throw an exception if this is the case.

File: sdk/communication/azure-communication-email/src/main/java/com/azure/communication/email/EmailClientBuilder.java [hunk 4]
```diff
@@ -240,6 +266,18 @@ private AzureCommunicationServicesClientImpl buildInnerClient() {
...
 
+    private HttpPipelinePolicy createHttpPipelineAuthPolicy() {
+        if (this.tokenCredential != null) {
+            return new BearerTokenAuthenticationPolicy(
+                this.tokenCredential, "https://communication.azure.com//.default");
+        } else if (this.azureKeyCredential != null) {
+            return new HmacAuthenticationPolicy(this.azureKeyCredential);
+        } else {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("Missing credential information while building a client."));
```
Comment (vcolin7) [inline sdk/communication/azure-communication-email/src/main/java/com/azure/communication/email/EmailClientBuilder.java:277]: An `IllegalStateException` is probably better here. We probably should also tell users how to remedy the problem (i.e. use one of the `credential()` methods to set a credential object). Any thoughts on this @srnagar?
```suggestion
                new IllegalStateException("Missing credential information while building a client."));
```

**Review by vcolin7 — APPROVED** [review]

**Pull request merged**